core/templates/pages/story-editor-page/navbar/story-editor-navbar.directive.ts

```diff
@@ -85,31 +81,11 @@ angular.module('oppia').directive('storyEditorNavbar', [
                   'Some chapters don\'t have exploration IDs provided.');
               }
             }
-
-            ExplorationSummaryBackendApiService.loadPublicExplorationSummaries(
-              explorationIds).then(function(summaries) {
-              if (summaries.length !== explorationIds.length) {
-                $scope.validationIssues.push(
-                  'Some explorations in story are not published.');
-              } else if (summaries.length > 0) {
-                var commonExpCategory = summaries[0].category;
-                for (var idx in summaries) {
-                  if (summaries[idx].category !== commonExpCategory) {
-                    $scope.validationIssues.push(
-                      'The explorations with IDs ' + summaries[0].id + ' and ' +
-                      summaries[idx].id + ' have different categories.'
-                    );
-                    break;
-                  }
-                  if (summaries[idx].language_code !== 'en') {
-                    $scope.validationIssues.push(
-                      'The explorations with ID ' + summaries[idx].id +
-                      ' is not in English.'
-                    );
-                    break;
-                  }
-                }
-              }
+            EditableStoryBackendApiService.validateExplorations(
```

Comment:
How often is this method called in practice? If it's a passive thing then this could result in lots of pings to the backend...

Reply:
This is only called whenever a change is applied, so shouldn't be too many times, right?

Reply:
Hm -- it seems a bit wasteful since the validation process is expensive. Could you do it only when the change relates to the list of explorations, and, in that case, do it only for the specific new exploration ID which is added or changed?

Reply:
I'll do it for when there are exploration ID changes, but the exploration category check requires all the explorations to be passed, so all exp IDs would have to be passed anyway, but can be restricted to only when any exp ID is changed.

Reply:
Done